src/workflows/recipes/components/FilterObjectForm.js

```diff
@@ -0,0 +1,803 @@
+  { value: 'stableSample', label: 'Stable' },
+  { value: '', label: 'None' },
+];
+
+/* Return a Map from a List.
+  Each item in the list is expected to be something like this:
+
+      [{"type": "age", "ages": [1,2,3]}, ]
+
+  Which is then transformed to this:
+
+      {"ages": [1,2,3]}
+
+  The list of valid keys needs to be known in advance. */
+export const serializeFilterObjectToMap = list => {
+  if (!list) {
+    return Map();
+  }
+  if (!List.isList(list)) {
+    throw new Error(`'${list}' (${typeof list}) is not a List`);
+  }
+
+  return Map(
+    list.map(item => {
+      const type = item.get('type');
+      if (type === 'bucketSample' || type === 'stableSample') {
+        return ['_sampling', item];
+      }
+      const name = MAPPING[type];
+      if (name === undefined) {
+        throw new Error(`Not sure how to convert '${type}'`);
+      }
+      return [name, item.get(name)];
+    }),
+  );
+};
+
+/* Return a List from a Map.
+  Each item in the map is expected to be key'ed based on the reverse of the MAPPING
+  constant. For example, from a list like this:
+
+       {locales: ["sv", "en-US"], ...}
+
+  Transforms to this:
+
+    [{type: "locale", locales: ["sv", "en-US"]}, ...]
+
+  The list of keys have to be known in advance. */
+export const deserializeFilterObjectToList = obj => {
+  if (!obj) {
+    return List();
+  }
+  const reverseMapping = Object.assign(
+    {},
+    ...Object.entries(MAPPING).map(([key, value]) => ({ [value]: key })),
+  );
+  return fromJS(
+    Object.entries(obj)
+      .filter(([key, value]) => {
+        return value && ((key === '_sampling' && value.type) || !!value.length);
+      })
+      .map(([key, value]) => {
+        if (key === '_sampling') {
+          // Special one that needs to be rewritten based on all the keys in the value.
+          // The 'value' should have
+          return Object.assign({}, value);
+        } else {
+          const name = reverseMapping[key];
+          if (name === undefined) {
+            throw new Error(`Key '${key}' no in known mapping.`);
+          }
+          return { [key]: value, type: name };
+        }
+      }),
+  );
+};
+
+class FilterObjectForm extends React.PureComponent {
+  static propTypes = {
+    allChannels: PropTypes.instanceOf(List),
+    allCountries: PropTypes.instanceOf(List),
+    allLocales: PropTypes.instanceOf(List),
+    disabled: PropTypes.bool,
+    filterObject: PropTypes.instanceOf(Map),
+  };
+
+  static defaultProps = {
+    allChannels: new List(),
+    allCountries: new List(),
+    allLocales: new List(),
+  };
+
+  allChannelOptions = () => {
+    return this.props.allChannels
+      .map(channel => {
+        return { label: channel.get('value'), value: channel.get('key') };
+      })
+      .toArray();
+  };
+
+  rememberActiveTabKey = key => {
+    window.sessionStorage.setItem('filterObjectActiveTabKey', key);
+  };
+
+  getDefaultActiveTabKey = (default_ = 'geo') => {
```

Comment:
Geo is likely the least used type of filter. Browser or sampling would would be a much better default.

Reply:
I can easily rearrange the order. But I think the left-most one should be visible by default. 

So, can you confirm this preferred order?:

1. Browser
2. Sampling
3. Geo